src/components/CreateItem.jsx

```diff
@@ -9,19 +10,45 @@ function CreateItem() {
     frequency: '7',
   };
   const [item, setItem] = useState(itemObj);
+  const token = localStorage.getItem('token');
+
+  const [value] = useCollection(
+    db.collection('items').where('token', '==', token),
+  );
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    const token = localStorage.getItem('token');
-
-    db.collection('items').add({
-      name: item.itemName,
-      frequency: item.frequency,
-      lastPurchasedDate: null,
-      date: Date().toLocaleString(),
-      token,
-    });
-    setItem(itemObj);
+
+    const cleanUserInput = (itemName) => {
+      return itemName.replace(/[\W_]+/g, '').toLowerCase();
+    };
+
+    const findDuplicateItem = (item) => {
+      let existingItem = false;
+
+      value.docs.forEach((doc) => {
+        const nameFromDb = doc.data().name;
+        const nameFromUser = item.itemName;
+
+        if (cleanUserInput(nameFromDb) === cleanUserInput(nameFromUser)) {
+          existingItem = true;
+        }
+      });
+      return existingItem;
+    };
```

Comment:
a very clean way to implement this functionality! nice separation of concern with the helper function too. Nice job team!👍